packages/lodestar/src/api/impl/beacon/blocks/utils.ts

```diff
@@ -52,29 +52,37 @@ async function resolveBlockIdOrNull(
   }
 
   if (blockId === "finalized") {
-    return await db.blockArchive.get(forkChoice.getFinalizedCheckpoint().epoch);
+    return await db.blockArchive.get(forkChoice.getFinalizedBlock().slot);
   }
 
+  let blockSummary;
+  let getBlockByBlockArchive;
+
   if (blockId.startsWith("0x")) {
-    const root = fromHexString(blockId);
-    const summary = forkChoice.getBlock(root);
-    if (summary) {
-      return await db.block.get(summary.blockRoot);
-    } else {
-      return await db.blockArchive.getByRoot(root);
+    const blockHash = fromHexString(blockId);
+    blockSummary = forkChoice.getBlock(blockHash);
```

Comment:
isn't this gonna return null for finalized blocks?

Reply:
Yeaj, it will be null for finalized blocks (except _the_ finalized block).
See the comments lower in the function that describe what should happen.